core/trino-main/src/test/java/io/trino/operator/TestHashAggregationOperator.java

```diff
@@ -698,6 +713,77 @@ public void testMemoryTracking()
         assertEquals(driverContext.getRevocableMemoryUsage(), 0);
     }
 
+    @Test
+    public void testAdaptivePartialAggregation()
+            throws Exception
+    {
+        List<Integer> hashChannels = Ints.asList(0);
+
+        HashAggregationOperatorFactory operatorFactory = new HashAggregationOperatorFactory(
+                0,
+                new PlanNodeId("test"),
+                ImmutableList.of(BIGINT),
+                hashChannels,
+                ImmutableList.of(),
+                PARTIAL,
+                ImmutableList.of(LONG_MIN.createAggregatorFactory(PARTIAL, ImmutableList.of(0), OptionalInt.empty())),
+                Optional.empty(),
+                Optional.empty(),
+                100,
+                Optional.of(DataSize.ofBytes(1)), // this setting makes operator to flush after each page
+                joinCompiler,
+                blockTypeOperators,
+                // use 5 rows threshold to trigger adaptive partial aggregation after each page flush
+                new PartialAggregationControllerFactory(true, 5, 0.8));
```

Comment:
Also add a test (or change this one) that there needs to be at least one flush before PA is disabled (E.g. low min row count, and flush happening after second input page.

Reply:
Added separate test case as this requires using `HashAggregationOperatorFactory` with a different `maxPartialMemory` limit